docs/streams/core-concepts.html

```diff
@@ -95,14 +95,14 @@
         The choice between event-time and ingestion-time is actually done through the configuration of Kafka (not Kafka Streams): From Kafka 0.10.x onwards, timestamps are automatically embedded into Kafka messages. Depending on Kafka's configuration these timestamps represent event-time or ingestion-time. The respective Kafka configuration setting can be specified on the broker level or per topic. The default timestamp extractor in Kafka Streams will retrieve these embedded timestamps as-is. Hence, the effective time semantics of your application depend on the effective Kafka configuration for these embedded timestamps.
     </p>
     <p>
-        Kafka Streams assigns a <b>timestamp</b> to every data record
-        via the <code>TimestampExtractor</code> interface.
-        Concrete implementations of this interface may retrieve or compute timestamps based on the actual contents of data records such as an embedded timestamp field
-        to provide event-time semantics, or use any other approach such as returning the current wall-clock time at the time of processing,
-        thereby yielding processing-time semantics to stream processing applications.
-        Developers can thus enforce different notions of time depending on their business needs. For example,
-        per-record timestamps describe the progress of a stream with regards to time (although records may be out-of-order within the stream) and
-        are leveraged by time-dependent operations such as joins.
+        Kafka Streams assigns a <b>timestamp</b> to every data record via the <code>TimestampExtractor</code> interface.
+        These per-record timestamps describe the progress of a stream with regards to time and are leveraged by	and are leveraged by time-dependent operations such as window operations.
+        As a result, this time will only advance when there is new record arrived at the processor.
```

Comment:
nit: "when a new record arrives at the processor"

Reply:
ack